Mathlib/CategoryTheory/ConnectedComponents.lean

```diff
@@ -40,6 +40,18 @@ def ConnectedComponents (J : Type u₁) [Category.{v₁} J] : Type u₁ :=
 instance [Inhabited J] : Inhabited (ConnectedComponents J) :=
   ⟨Quotient.mk'' default⟩
 
+/-- Every function from connected components of a category gives a functor to discrete category -/
+def ConnectedComponents.connectedToDiscrete.{u_1, u_2} {X C : Type u_1} [Category.{u_2, u_1} C]
+    (f : ConnectedComponents C -> X) : (C ⥤ Discrete X) where
+  obj :=  Discrete.mk ∘ f ∘ Quotient.mk (Zigzag.setoid _)
+  map :=  Discrete.eqToHom ∘ congrArg f ∘ Quotient.sound ∘ Zigzag.of_hom
```

Comment:
```suggestion
    (f : ConnectedComponents C → X) : C ⥤ Discrete X where
  obj Y :=  Discrete.mk (f (Quotient.mk (Zigzag.setoid _) Y))
  map g := Discrete.eqToHom (congrArg f (Quotient.sound (Zigzag.of_hom g)))
```
- You generally see a lot more unicode arrows in mathlib instead of `->`, but I don't know if this is a strict rule.
- I think it's a bit clearer to see what a functor does to objects/morphisms if you use my suggested notation instead of `∘`.